frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java

```diff
@@ -39,7 +39,7 @@ public VcsRootComboBox(@MinLen(1) List<GitRepository> repositories) {
 
   @Override
   @UIEffect
-  public void updateRepositories(@MinLen(1) List<GitRepository> repositories) {
+  public synchronized void updateRepositories(@MinLen(1) List<GitRepository> repositories) {
```

Comment:
Changes to this file are related neither to #190 nor to #252. Those are to avoid another error in a case when `getSelectedRepository` is called during `updateRepositories` (exactly when previous repos was removed but new ones are not added yet)... :exploding_head:

Reply:
Undone on #257 :/ Rule of the thumb:
* Locks (and many other concurrency primitives) suck :banana: 
* Futures (or more generally: schedulable tasks) rock :rocket: